json-schema-core/src/main/java/io/ballerina/jsonschema/core/Generator.java

```diff
@@ -0,0 +1,254 @@
+                    + DEFAULT_SCHEMA_NAME + WHITE_SPACE + generatedType + SEMI_COLON;
+            ModuleMemberDeclarationNode schemaNode = NodeParser.parseModuleMemberDeclaration(schemaDefinition);
+            this.nodes.put(DEFAULT_SCHEMA_NAME, schemaNode);
+        }
+
+        ModulePartNode modulePartNode = generateModulePartNode();
+        String generatedTypes = formatModuleParts(modulePartNode);
+        return new Response(generatedTypes, this.diagnostics);
+    }
+
+    public String convert(Object schemaObject, String name) {
+        if (schemaObject instanceof Boolean) {
+            boolean boolValue = (Boolean) schemaObject;
+            return boolValue ? JSON : NEVER;
+        }
+
+        Schema schema = (Schema) schemaObject;
+
+        ArrayList<Object> schemaType = getCommonType(schema.getEnumKeyword(), schema.getConstKeyword(),
+                schema.getType());
+        if (schemaType.isEmpty()) {
+            ID_TO_TYPE_MAP.put(schema.getIdKeyword(), NEVER);
+            return NEVER;
+        } else if (schemaType.contains(Class.class)) {
+            schemaType.remove(Class.class);
+            if (schemaType.size() == 1) {
+                String typeName = createType(name, schema, schemaType.getFirst(), this);
+                ID_TO_TYPE_MAP.put(schema.getIdKeyword(), typeName);
+                return typeName;
+            } else {
+                Set<String> unionTypes = new HashSet<>();
+                for (Object element : schemaType) {
+                    String subtypeName = name + getBallerinaType(element);
+                    unionTypes.add(createType(subtypeName, schema, element, this));
+                }
+                if (unionTypes.containsAll(
+                        Set.of(INTEGER, NUMBER, BOOLEAN, STRING, UNIVERSAL_ARRAY, UNIVERSAL_OBJECT, NULL))) {
+                    ID_TO_TYPE_MAP.put(schema.getIdKeyword(), JSON);
+                    return JSON;
+                }
+                if (unionTypes.contains(NUMBER)) {
+                    unionTypes.remove(NUMBER);
+                    unionTypes.add(INTEGER);
+                    unionTypes.add(FLOAT);
+                    unionTypes.add(DECIMAL);
+                }
+                String typeName = String.join(PIPE, unionTypes);
+                ID_TO_TYPE_MAP.put(schema.getIdKeyword(), typeName);
+                return typeName;
+            }
+        } else {
+            //TODO: Validate constraints on enums
+            String typeName = schemaType.stream().map(String::valueOf).collect(Collectors.joining(PIPE));
+            ID_TO_TYPE_MAP.put(schema.getIdKeyword(), typeName);
+            return typeName;
+        }
+    }
+
+    public static ArrayList<Object> getCommonType(ArrayList<Object> enumKeyword, Object constKeyword,
+                                                  ArrayList<String> type) {
+        Set<Object> typeList = new HashSet<>();
+        Set<Object> finalList = new HashSet<>();
```

Comment:
We can strongly type these right?

```suggestion
        Set<Class<?>> typeList = new HashSet<>();
```